app/furniture/marketplace/delivery_areas_controller.rb

```diff
@@ -32,7 +32,11 @@ def update
     end
 
     def destroy
-      authorize(delivery_area).destroy
+      authorize(delivery_area)
+      # rubocop:disable Rails/SkipsModelValidations
+      delivery_area.carts.update_all(delivery_area_id: nil)
+      # rubocop:enable Rails/SkipsModelValidations
+      delivery_area.destroy
```

Comment:
Might be worth wrapping these two lines in a transaction.

Reply:
ooo good point.

Reply:
Random thought: could we accomplish this with something like this on the Delivery Area model?

```ruby
  has_many :pre_checkout_carts, ...appropriate scopes go here..., dependent: :nullify
```